core/nats-macros/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+use darling::{FromDeriveInput, FromMeta};
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{self, parse_macro_input, DeriveInput};
+
+#[derive(FromDeriveInput)]
+#[darling(attributes(typed_message))]
+struct Opts {
+    subject: NatsSubjectMacroInvocation,
+    #[darling(default)]
+    response: ResponseType,
+}
+
+#[derive(FromMeta, Clone, Debug)]
+struct ResponseType(syn::Type);
+
+impl Default for ResponseType {
+    fn default() -> Self {
+        ResponseType(syn::Type::from_string("NoReply").unwrap())
+    }
+}
+
+#[derive(Debug)]
+struct NatsSubjectMacroInvocation(syn::Macro);
+
+impl FromMeta for NatsSubjectMacroInvocation {
+    fn from_string(value: &str) -> darling::Result<Self> {
+        let props: Vec<String> = value
+            .split('.')
+            .map(|ea| ea.strip_prefix('#'))
+            .filter(|ea| ea.is_some())
```

Comment:
I think lines 31, 32, 33 could all be replaced with one `.filter_map(|ea| ea.strip_prefix('#'))`